go.mod

```diff
@@ -295,3 +296,6 @@ replace github.com/benbjohnson/clock v1.3.0 => github.com/benbjohnson/clock v1.1
 
 // copy from TiDB
 replace go.opencensus.io => go.opencensus.io v0.23.1-0.20220331163232-052120675fac
+
+// todo: remove this replace after https://github.com/segmentio/kafka-go/pull/1059 merged
+replace github.com/segmentio/kafka-go v0.4.38 => github.com/3AceShowHand/kafka-go v0.0.0-20230109084533-cd584ac8c705
```

Comment:
Do we have to rely on the PR?
What about using `github.com/pingcap/kafka-go`?

Reply:
> Do we have to rely on the PR? What about using `github.com/pingcap/kafka-go`?

The master branch of the kafka-go does not support our desired feature (allow customizable field) at the moment, so we decide to use our development branch temporarily.